uniffi_bindgen/src/bindings/kotlin/gen_kotlin.rs

```diff
@@ -60,241 +76,294 @@ impl MergeWith for Config {
-            Type::Float64 => "Double".to_string(),
-            // These types need conversion, and special handling for lifting/lowering.
-            Type::Boolean => "Boolean".to_string(),
-            Type::String => "String".to_string(),
-            Type::Timestamp => "java.time.Instant".to_string(),
-            Type::Duration => "java.time.Duration".to_string(),
-            Type::Enum(name)
-            | Type::Record(name)
-            | Type::Object(name)
-            | Type::Error(name)
-            | Type::CallbackInterface(name) => class_name_kt(name)?,
-            Type::Optional(t) => format!("{}?", type_kt(t)?),
-            Type::Sequence(t) => format!("List<{}>", type_kt(t)?),
-            Type::Map(t) => format!("Map<String, {}>", type_kt(t)?),
-            Type::External { .. } => panic!("no support for external types yet"),
-            Type::Wrapped { .. } => panic!("no support for wrapped types yet"),
-        })
+    pub fn initialization_code(&self) -> Vec<String> {
+        let oracle = &self.oracle;
+        self.members()
+            .into_iter()
+            .filter_map(|member| member.initialization_code(oracle))
+            .collect()
     }
 
-    /// Get the Kotlin syntax for representing a given low-level `FFIType`.
-    pub fn type_ffi(type_: &FFIType) -> Result<String, askama::Error> {
-        Ok(match type_ {
-            // Note that unsigned integers in Kotlin are currently experimental, but java.nio.ByteBuffer does not
-            // support them yet. Thus, we use the signed variants to represent both signed and unsigned
-            // types from the component API.
-            FFIType::Int8 | FFIType::UInt8 => "Byte".to_string(),
-            FFIType::Int16 | FFIType::UInt16 => "Short".to_string(),
-            FFIType::Int32 | FFIType::UInt32 => "Int".to_string(),
-            FFIType::Int64 | FFIType::UInt64 => "Long".to_string(),
-            FFIType::Float32 => "Float".to_string(),
-            FFIType::Float64 => "Double".to_string(),
-            FFIType::RustArcPtr => "Pointer".to_string(),
-            FFIType::RustBuffer => "RustBuffer.ByValue".to_string(),
-            FFIType::ForeignBytes => "ForeignBytes.ByValue".to_string(),
-            FFIType::ForeignCallback => "ForeignCallback".to_string(),
-        })
+    pub fn declaration_code(&self) -> Vec<String> {
+        let oracle = &self.oracle;
+        self.members()
+            .into_iter()
+            .filter_map(|member| member.definition_code(oracle))
+            .chain(
+                self.ci
+                    .iter_types()
+                    .into_iter()
+                    .filter_map(|type_| oracle.find(&type_).helper_code(oracle)),
+            )
+            .collect()
+    }
+
+    pub fn imports(&self) -> Vec<String> {
+        let oracle = &self.oracle;
+        let mut imports: Vec<String> = self
+            .members()
+            .into_iter()
+            .filter_map(|member| member.import_code(oracle))
```

Comment:
I've renamed the `CodeType` and `CodeDeclaration` methods as `imports`, since the templates are rendering the import _statements_.

I was thinking about import aliases when writing this as `import_code`, but I don't think that will pan out:

e.g. `import JSONObject as _ImportedJSONObject`.

Reply:
That actually works great,[ I used it to implement `External`.](https://github.com/mozilla/uniffi-rs/pull/1046/files/1553062784d85a5b5bf7267a7f3ade4ea70a5ae4#diff-7db1b800084cb40cbbd4343d949272a8e3b8972d6c8f4a607fd7c73f8f58897aR42-R48)